app/com/arpnetworking/metrics/portal/scheduling/impl/PeriodicSchedule.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2018 Dropbox, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.arpnetworking.metrics.portal.scheduling.impl;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import models.internal.scheduling.BaseSchedule;
+import net.sf.oval.constraint.NotNull;
+import net.sf.oval.constraint.ValidateWithMethod;
+
+import java.time.Duration;
+import java.time.ZonedDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.Optional;
+
+/**
+ * Schedule for a job that repeats periodically.
+ *
+ * @author Spencer Pearson (spencerpearson at dropbox dot com)
+ */
+public final class PeriodicSchedule extends BaseSchedule {
+
+    /**
+     *
+     */
+    public enum Period {
```

Comment:
The current convention in ArpNetworking bucks Java convention that nested types are listed after all other elements.